agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/GlobalVariables.kt

```diff
@@ -17,13 +17,12 @@ package com.epam.drill.agent.configuration
 
 import kotlin.native.concurrent.*
 import com.epam.drill.common.agent.configuration.*
-import com.epam.drill.transport.*
 
 private val _requestPattern = AtomicReference<String?>(null).freeze()
 private val _drillInstallationDir = AtomicReference<String?>(null).freeze()
 private val _adminAddress = AtomicReference<URL?>(null).freeze()
-private val _secureAdminAddress = AtomicReference<URL?>(null).freeze()
 private val _agentConfig = AtomicReference<AgentConfig?>(null).freeze()
+private val _agentParameters = AtomicReference(AgentParameters().freeze()).freeze()
```

Comment:
According to [Kotlin docs](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native.concurrent/freeze.html) freeze makes the whole subgraph immutable.
So calling freeze on reference should be enough?
```
private val _agentParameters = AtomicReference(AgentParameters()).freeze()
```

Reply:
It doesn't work without this freeze() by some reason.

Reply:
Interesting, but I guess we can live with that 😃